### DISCUSSION TITLE
Support for reading .mat and .hdf5 files

### DISCUSSION THREAD

**Discussion by saravanabalagi** [visgl/loaders.gl#2315] [Ideas]
I often find myself needing to read .mat (MATLAB files) and [HDF5](https://www.hdfgroup.org/solutions/hdf5/) files for visualization purposes. I think it would be a good idea to add support for them as part of the awesome lib!

**Comment by ibgreen** [comment]
Thanks for the kind words and for the feedback. Those formats could indeed be valuable additions. 

While it wouldn't be a top priority for current maintainers, it doesn't hurt to discuss. It would be much easier if we found some existing libraries to build upon.

- Unfortunately https://github.com/KovacsGG/mat4js is GPL v3 licensed which does not fit the philosophy of loaders.gl.
- However https://github.com/usnistgov/jsfive/blob/master/LICENSE.txt seems suitable for HDF5.

If I understand it correctly, the most recent versions of .mat files are HDF5 based (see the final comment in the mat4js README.
